File: src/windows/magma/app/compile/ConstructorHeader.c
#include "./ConstructorHeader.h"
export class ConstructorHeader implements MethodHeader<ConstructorHeader> {
	mut generateWithAfterName(afterName: &[I8]): &[I8] {
		return "constructor " + afterName;
	}
	mut hasAnnotation(annotation: &[I8]): Bool {
		return false;
	}
	mut removeModifier(modifier: &[I8]): ConstructorHeader {
		return this;
	}
	mut addModifier(modifier: &[I8]): ConstructorHeader {
		return this;
	}
}

File: src/java/magma/app/compile/define/ConstructorHeader.java
package magma.app.compile.define;

public class ConstructorHeader implements MethodHeader {
    @Override
    public String generateWithAfterName(String afterName) {
        return "constructor " + afterName;
    }

    @Override
    public boolean hasAnnotation(String annotation) {
        return false;
    }

    @Override
    public MethodHeader removeModifier(String modifier) {
        return this;
    }
}
